src/ui/common/src/reducers/workflow.ts

```diff
@@ -140,22 +166,91 @@ export const handleGetWorkflow = createAsyncThunk<
   }
 );
 
-const handleSelectResultIdx = (state: WorkflowState, idx: number) => {
-  state.artifactResults = {};
-  state.operatorResults = {};
-  state.selectedResult = state.dagResults[idx];
-  state.selectedDag = state.dags[state.selectedResult.workflow_dag_id];
-};
+export const handleGetSelectDagPosition = createAsyncThunk<
+  positionResponse,
+  {
+    apiKey: string;
+    operators: { [id: string]: Operator };
+    artifacts: { [id: string]: Artifact };
+  }
+>(
+  'workflowReducer/getSelectDagPosition',
+  async (
+    args: {
+      apiKey: string;
+      operators: { [id: string]: Operator };
+      artifacts: { [id: string]: Artifact };
+      onChange: () => void;
+      onConnect: (any) => void;
+    },
+    thunkAPI
+  ) => {
+    const { apiKey, operators, artifacts, onChange, onConnect } = args;
+    const res = await fetch(`${apiAddress}/api/positioning`, {
+      method: 'POST',
+      headers: {
+        'api-key': apiKey,
+      },
+      body: JSON.stringify(operators),
+    });
+
+    const position = await res.json();
+    if (!res.ok) {
+      return thunkAPI.rejectWithValue(position.error);
+    }
+
+    const opPositions = position.operator_positions;
+    const artfPositions = position.artifact_positions;
+    const opNodes = Object.values(operators)
+      .filter((op) => {
+        return op.spec.type != OperatorType.Param;
+      })
+      .map((op) =>
+        getOperatorNode(op, opPositions[op.id], onChange, onConnect)
+      );
+    const artfNodes = Object.values(artifacts).map((artf) =>
+      getArtifactNode(artf, artfPositions[artf.id], onChange, onConnect)
+    );
+    const edges = getEdges(operators);
+    return {
+      nodes: opNodes.concat(artfNodes),
+      edges: edges,
+    } as positionResponse;
+  }
+);
 
 export const workflowSlice = createSlice({
   name: 'workflowReducer',
   initialState,
   reducers: {
     selectResultIdx: (state, { payload }: PayloadAction<number>) => {
-      handleSelectResultIdx(state, payload);
+      state.artifactResults = {};
```

Comment:
https://www.loom.com/share/dbb2c6def3174031a75697ec274e9cbb
Explains why I need to set the state directly instead of calling a function